test/FunctionalTests/Microsoft.OData.Core.Tests/Microsoft.OData.Core.Tests.csproj

```diff
@@ -19,6 +19,10 @@
     <OutputPath>..\..\..\bin\AnyCPU\Release\Test\</OutputPath>
     <DefineConstants>TRACE</DefineConstants>
   </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+    <DefineConstants>$(DefineConstants);NETCOREAPP3_1_OR_GREATER;NETCOREAPP;NETCOREAPP3_1</DefineConstants>

```

Comment:
Why was this necessary?

Well, in the library and test code bases, I used `NETCOREAPP3_1_OR_GREATER` to conditionally compile code only for .NET Core 3.1+. However, it turns out this constant is not defined by default when you're building the project using .NET Core 3.1 SDK, it's only available by default with .NET 5 SDK and greater. So that's why I manually defined this constant. the `NETCOREAPP3_1` and `NETCOREAPP` symbols are defined by default in .NET Core 3.1, so I didn't need add them here. But I just added them for completeness because, as it turns out, I had already done the same thing in the library `Microsoft.OData.Core.csproj` project file.

The tests were failing to build in the CI pipeline because it's using .NET Core 3.1 SDK to build the tests, so the code under the `NETCOREAPP3_1_OR_GREATER` symbols was not getting compiled. Not it works. This has also made me realize that none of the tests under this symbol (including the `ODataUtf8JsonWriter` tests) ever ran in our build pipeline. We just never noticed. 

Check out [this guide](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives) to see which compiler symbols are auto-defined in the different framework versions.

Reply:
This is weird that there are some tests that might not be running on the pipeline and we just don't realize.

Reply:
Yes. Though I do not have hard evidence that the tests did not run. Looking at past logs, I don't see detailed break down of all the tests that ran.